apps/sms/services/js/drafts.js

```diff
@@ -154,9 +152,14 @@
      * Stores drafts that are held in memory to local storage.
      */
     store: function() {
+      var defer = Utils.Promise.defer();
+
       asyncStorage.setItem('draft index', [...draftIndex], () => {
         InterInstanceEventDispatcher.emit('drafts-changed');
+        defer.resolve();
       });
+
+      return defer.promise;
```

Comment:
reminds me of something... oh yeah, I know :) https://github.com/mozilla-b2g/gaia/pull/30898/files#diff-c565d87c361e6319685488ffcca9557c